draft-ietf-quic-transport.md

```diff
@@ -1847,6 +1847,11 @@ endpoint migrates to a new address.
 An endpoint MUST NOT initiate connection migration before the handshake is
 finished and the endpoint has 1-RTT keys.  The design of QUIC relies on
 endpoints retaining a stable address for the duration of the handshake.
+Clients MUST NOT initiate connection migration before they are
+certain that their peer also considers the handshake finished. This
+means that in addition to waiting for availability of 1-RTT keys,
+clients MUST wait acknowledgement by the server of one of their
+1-RTT messages before initiating connection migration.
 
 An endpoint also MUST NOT initiate connection migration if the peer sent the
 `disable_migration` transport parameter during the handshake.  An endpoint which
```

Comment:
Not in this PR, but why do we use both client, server, peer, and endpoint, when migration can only happen from the client in V1 (or did I miss something?)

Reply:
Some of that's just a mix of different text coming in at different times, I think. 
At one point, the idea was to deliberately keep the text agnostic (for the most part, any text restricting who can initiate is clearly going to be one-sided) to endpoint, since that allowed reasoning about it as initiator and responder rather than client vs. server.

Reply:
The part about 1-rtt ack is actually client specific. I think I was influenced by the picotls implementation that only provides the 1-rtt read key after CFIN has been received, and did not spell out the parallel server requirement that some 1rtt packet should have been received. Will do.

Reply:
@huitema  > Where did we specify that only the client initiates migrations in V1?

I'm not sure, but from memory it was dropped at some point because symmetric ping pong got hairy during updates or some defence became difficult.

Maybe it didn't land? Found this:

> Caution:
If both endpoints change connection ID in response to seeing a change in connection ID from their peer, then this can trigger an infinite sequence of changes.

But there is this, which seems to be preferred address only - maybe I got it mixed up_

9.6, transport:
> Migrating a connection to a new server address mid-connection is left for future work.